pkg/storage/storage.go

```diff
@@ -115,7 +116,15 @@ func (p *Storage) GetContainerMetrics(pods ...apitypes.NamespacedName) ([]api.Ti
 	return timestamps, resMetrics
 }
 
+func (p *Storage) IsEmpty() bool {
+	return len(p.nodes) == 0
+}
+
 func (p *Storage) Store(batch *MetricsBatch) error {
+	// No node has been scraped. Return so the previous updated cache remains intact
+	if len(batch.Nodes) == 0 {
+		return fmt.Errorf("no nodes found in the input batch")
+	}
 	newNodes := make(map[string]NodeMetricsPoint, len(batch.Nodes))
```

Comment:
Scraper sends empty batch when scrape has no nodes. Guarding against it using parameter validation.

Do we need to invalidate the cache as data can become stale though its a larger problem that scrape isnt going through for longer duration

Reply:
Cache invalidation was not discussed in proposal. Let's try to fix one problem at once

Reply:
`if len(batch.Nodes) == 0 {
		return fmt.Errorf("no nodes found in the input batch")
	}`

Is this check required or to be removed ?